ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/commons/objective_function_evaluator/ObjectiveFunction.java

```diff
@@ -48,59 +50,47 @@ public Set<FlowCnec> getFlowCnecs() {
+        return functionalCostEvaluator.computeCostAndLimitingElements(flowResult, rangeActionActivationResult, sensitivityResult, sensitivityStatus, contingenciesToExclude);
     }
 
-    public List<FlowCnec> getMostLimitingElements(FlowResult flowResult, RangeActionActivationResult rangeActionActivationResult, SensitivityResult sensitivityResult, int number) {
-        return functionalCostEvaluator.getCostlyElements(flowResult, rangeActionActivationResult, sensitivityResult, number);
-    }
-
-    public double getVirtualCost(FlowResult flowResult,  RangeActionActivationResult rangeActionActivationResult, SensitivityResult sensitivityResult, ComputationStatus sensitivityStatus) {
-        return virtualCostEvaluators.stream()
-                .mapToDouble(costEvaluator -> costEvaluator.computeCost(flowResult, rangeActionActivationResult, sensitivityResult, sensitivityStatus))
-                .sum();
+    public Set<String> getVirtualCostNames() {
+        return virtualCostEvaluators.stream().map(CostEvaluator::getName).collect(Collectors.toSet());
     }
 
-    public double getVirtualCost(FlowResult flowResult,  RangeActionActivationResult rangeActionActivationResult, SensitivityResult sensitivityResult, ComputationStatus sensitivityStatus, Set<String> contingenciesToExclude) {
-        return virtualCostEvaluators.stream()
-            .mapToDouble(costEvaluator -> costEvaluator.computeCost(flowResult, rangeActionActivationResult, sensitivityResult, sensitivityStatus, contingenciesToExclude))
-            .sum();
+    public  Pair<Double, List<FlowCnec>> getVirtualCostAndCostlyElements(FlowResult flowResult, RangeActionActivationResult rangeActionActivationResult, SensitivityResult sensitivityResult, ComputationStatus sensitivityStatus) {
+        return getVirtualCostAndCostlyElements(flowResult, rangeActionActivationResult, sensitivityResult, sensitivityStatus, new HashSet<>());
     }
 
-    public Set<String> getVirtualCostNames() {
-        return virtualCostEvaluators.stream().map(CostEvaluator::getName).collect(Collectors.toSet());
+    public  Pair<Double, List<FlowCnec>> getVirtualCostAndCostlyElements(FlowResult flowResult, RangeActionActivationResult rangeActionActivationResult, SensitivityResult sensitivityResult, ComputationStatus sensitivityStatus, Set<String> contingenciesToExclude) {
+        AtomicReference<Double> totalCost = new AtomicReference<>((double) 0);
+        List<FlowCnec> allCostlyElements = new ArrayList<>();
+        virtualCostEvaluators.forEach(costEvaluator -> {
+            Pair<Double, List<FlowCnec>> costAndCostlyElements = costEvaluator.computeCostAndLimitingElements(flowResult, rangeActionActivationResult, sensitivityResult, sensitivityStatus, contingenciesToExclude);
+            totalCost.updateAndGet(v -> v + costAndCostlyElements.getLeft());
+            allCostlyElements.addAll(costAndCostlyElements.getRight());
+        });
+        return Pair.of(totalCost.get(), allCostlyElements);
```

Comment:
you should sort allCostlyElements by descending virtual cost. otherwise delete this function (and the previous one) since it doesn't seem to be used